config/from_test.go

```diff
@@ -21,18 +21,19 @@ func TestFromURLMatchesRequestURL(t *testing.T) {
 		{"https://from.example.com", "https://to.example.com/some/path", false},
 		{"https://*.example.com", "https://from.example.com/some/path", true},
 		{"https://*.example.com", "https://example.com/some/path", false},
+		{"https://*.example.com", "https://from.example.com:8443/some/path", true},
 	} {
 		fromURL := urlutil.MustParseAndValidateURL(tc.pattern)
 		requestURL := urlutil.MustParseAndValidateURL(tc.input)
-		assert.Equal(t, tc.matches, FromURLMatchesRequestURL(&fromURL, &requestURL),
+		assert.Equal(t, tc.matches, FromURLMatchesRequestURL(&fromURL, &requestURL, true),
 			"from-url: %s\nrequest-url: %s", tc.pattern, tc.input)
 	}
 }
 
 func TestWildcardToRegex(t *testing.T) {
 	t.Parallel()
 
-	re, err := regexp.Compile(WildcardToRegex("*.internal.*.example.com"))
+	re, err := regexp.Compile(WildcardToRegex("*.internal.*.example.com", true))
```

Comment:
Would it be worth adding a unit test exercise the `WildcardToRegex(..., false)` code path too?